knip.ts

```diff
@@ -0,0 +1,136 @@
+import type { KnipConfig } from 'knip';
+
+export default {
+  rules: {
+    classMembers: 'off',
+    duplicates: 'off',
+    enumMembers: 'off',
+    exports: 'off',
+    nsExports: 'off',
+    nsTypes: 'off',
+    types: 'off',
+    unresolved: 'off',
+  },
+  workspaces: {
+    '.': {
+      ignoreDependencies: [
+        '@babel/code-frame',
+        '@babel/core',
+        '@babel/eslint-parser',
+        '@babel/parser',
+        '@babel/types',
+        '@nx/workspace',
+        'cross-fetch',
+        'glob',
+        'husky',
+        'jest-specific-snapshot',
+        'make-dir',
+        'ncp',
+        'tmp',
+
+        // imported in eslint.config.js
+        '@typescript-eslint/utils',
+        // imported in eslint.config.mjs
+        '@typescript-eslint/eslint-plugin-internal',
+      ],
+      entry: ['tools/release/changelog-renderer.js'],
+      ignoreBinaries: [
+        // https://github.com/webpro/knip/issues/433
+        'stylelint',
+      ],
+    },
+    'packages/ast-spec': {
+      ignore: [
+        'src/**/fixtures/**',
+        'tests/*.type-test.ts',
+        // @typescript-eslint/typescript-estree is not listed in dependencies to avoid circular dependency errors
+        // You can check a more detailed explanation in this file
+        'tests/util/parsers/typescript-estree-import.ts',
+      ],
+    },
+    'packages/eslint-plugin': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/eslint-plugin-internal': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/integration-tests': {
+      ignore: ['fixtures/**'],
+    },
+    'packages/parser': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/repo-tools': {
+      ignoreDependencies: ['tsconfig.json'],
+    },
```

Comment:
Knip thinks that `tsconfig.json` is unlisted dependency.

We could change 
```diff
-    "typecheck": "npx tsc -p tsconfig.json --noEmit"
+    "typecheck": "tsc -p tsconfig.json --noEmit"
```

in 

https://github.com/typescript-eslint/typescript-eslint/blob/e44a1a280f08f9fd0d29f74e5c3e73b7b64a9606/packages/repo-tools/package.json#L16

to avoid this error.

But I'm not sure if we can safely do this because `npx` was intentionally added in https://github.com/typescript-eslint/typescript-eslint/pull/8450

Reply:
Yeah this feels like a feature request for Knip: shouldn't it understand that `npx tsc` goes to TypeScript?

...separately, `-p` defaults to `tsconfig.json`. `yarn knip` and `yarn typecheck` pass happily for me when I remove the `-p tsconfig.json` altogether. Maybe a good reason to remove them?